src/dumpContent.ts

```diff
@@ -0,0 +1,110 @@
+#!npx vite-node
+import fs from 'fs'
+import { promisify } from 'util'
+
+type ObjData = { _id: string }
+type SearchData = { continuation?: string; objs: ObjData[] }
+type BlobsData = { private_access: { get: { url: string } } }
+
+const DIRECTORY = 'contentDump'
+
+const API_KEY = process.env.SCRIVITO_API_KEY
+const INSTANCE_ID = process.env.SCRIVITO_TENANT
+
+if (INSTANCE_ID && API_KEY) {
+  clearDump()
+  await dumpContent()
+  console.log('\n✅ Dump complete.')
+} else {
+  console.error(
+    'Please provide SCRIVITO_TENANT and SCRIVITO_API_KEY of the instance to dump.',
+  )
+  process.exitCode = -1
+}
+
+function clearDump() {
+  fs.rmSync(DIRECTORY, { force: true, recursive: true })
+  fs.mkdirSync(`${DIRECTORY}/objs`, { recursive: true })
+  fs.mkdirSync(`${DIRECTORY}/binaries`, { recursive: true })
+}
+
+async function dumpContent() {
+  let continuation: string | undefined
+
+  do {
+    const data: SearchData = await fetchJson<SearchData>(
+      `workspaces/published/objs/search`,
+      JSON.stringify({
+        continuation,
+        include_objs: true,
+        options: { site_aware: true },
+        size: 10,
+      }),
+    )
+
+    for (const objData of data.objs) await dumpObjAndBinaries(objData)
+
+    continuation = data.continuation
+  } while (continuation)
+}
+
+async function dumpObjAndBinaries(objData: ObjData) {
+  await dumpBinaries(objData)
+  dumpObj(objData)
+}
+
+async function dumpBinaries(data: unknown) {
+  if (!data || typeof data !== 'object') return
+  if (isBinaryData(data)) await dumpBinary(data[1].id)
+  else for (const d of Object.values(data)) await dumpBinaries(d)
+}
```

Comment:
I have a hard time to understand this function. Now - after longer reading of the actual code and data structure of the objs - I think we should make it more explicit. I would explicitly handle `_widget_pool` and avoid recursion. Something like this:

```ts
// earlier in the file:
type ObjData = { _id: string; _widget_pool: { [key: string]: object } }

async function dumpBinaries(objData: ObjData) {
  const { _widget_pool, ...otherAttributes } = objData

  for (const widget of Object.values(_widget_pool)) {
    for (const widgetAttribute of Object.values(widget)) {
      await dumpBinaryAttribute(widgetAttribute)
    }
  }

  for (const objAttribute of Object.values(otherAttributes)) {
    await dumpBinaryAttribute(objAttribute)
  }
}

async function dumpBinaryAttribute(data: unknown) {
  if (!isBinaryAttribute(data)) return
  return dumpBinary(data[1].id)
}
```

WDYT?

Reply:
Binary attributes in widgets are very rare. Having different code paths for widget and obj attributes may lead to a situation where the widget code path is never taken. Since we have no tests, this may lead to bugs that stay undetected for very long.
I admit that expecting (searching for) binary attributes anywhere is a simplification but I don't think it's an oversimplification since the very specific type signature can not happen by chance.

I'll give it a try anyway.